app/src/main/java/com/android/sample/model/filter/Filter.kt

```diff
@@ -41,19 +32,15 @@ data class FloatRange(var min: Float, var max: Float, var minBorn: Float, var ma
   }
 
   fun update(newMin: Float, newMax: Float) {
-    require(0 <= newMin) { NEW_MIN_SHOULD_NOT_BE_NEGATIVE }
-    require(0 <= newMax) { NEW_MAX_SHOULD_NOT_BE_NEGATIVE }
-    require(newMin <= newMax) { NEW_MIN_SHOULD_NOT_EXCEED_MAX }
-    if (minBorn == UNINITIALIZED_BORN_VALUE && maxBorn == UNINITIALIZED_BORN_VALUE) {
-      minBorn = newMin.roundToInt().toFloat()
-      maxBorn = newMax.roundToInt().toFloat()
-    }
-    require(newMin >= minBorn && newMax <= maxBorn) { NEW_MIN_AND_NEW_MAX_SHOULD_BE_WITHIN_RANGE }
-    min = newMin.roundToInt().toFloat()
-    max = newMax.roundToInt().toFloat()
+    require(newMin <= newMax) { MIN_SHOULD_NOT_BE_GREATER_THAN_MAX }
+    min = newMin
+    max = newMax
   }
 
-  fun isLimited() = min == minBorn && max == maxBorn
+  fun isLimited() =
```

Comment:
**`isLimited` Logic:**  
   Great work simplifying the logic here! Maybe toss in a quick comment for clarity—it’s a bit tricky at first glance.

Reply:
It is done!